### ISSUE TITLE
Differentiate selection mode

### ISSUE THREAD

**Issue by hari-rangarajan** [thestinger/termite#241]
It would be nice to have a visual indicator to differentiate selection vs insert mode.

The simplest would be based on the following parameters:

1) fore-ground color
2) background color
3) mouse cursor shape

I can make a patch with some help on the coding style. My thinking would be to introduce additional config parameters for the above three options suffixed with "-select-mode".

**Comment by hari-rangarajan** [comment]
Initial attempt with background color
https://github.com/thestinger/termite/pull/242

**Comment by languitar** [comment]
:+1:

**Comment by fredlebel** [comment]
That would definitely be useful.  +1

**Comment by thestinger** [comment]
Once https://mail.gnome.org/archives/commits-list/2015-November/msg05845.html ends up in a stable release, it can be done by changing the cursor color.

**Comment by jelly** [comment]
Does this feature collide with #284 or should be referenced to this ticket.

**Comment by thestinger** [comment]
It depends on #284 being implemented. It can just be done in a way where selection mode has a different color configuration option.

**Comment by thestinger** [comment]
Termite is obsolete. Please use [Alacritty](https://github.com/alacritty/alacritty) instead. See https://github.com/thestinger/termite#termite-is-obsoleted-by-alacritty for more details.
